### ISSUE TITLE
Vendor list improvements

### ISSUE THREAD

**Issue by MariusJulius** [keeleinstituut/tv-tolkevarav#469]

**Comment by plakitkelly** [comment]
Steps:
1. Open vendor detail view
2. Scroll down to "Hinnakiri" section
3. Go to another page (not 1st)
4. Click on edit button in any language direction row 
5. Go to "Lisa oskused" tab and select any skill that is not selected
6. Set prices and click "Salvesta"

Actual: User doesn't see added skill in selected page. If user goes to 1st page, there is added skill in first row. Now I have this language direction on 1st and another page. If user wants to add more skills on first (or another) page, user cannot see that some skills are selected (this one what is on another page)
If user adds selected skill again, then "Salvesta" button does nothing.

Expected: Group same language direction skills and show selected skills in "Lisa oskused" tab

[VIDEO](https://drive.google.com/file/d/1osNQZv2_hOfiaBdP6Fuul7aah058Cx9N/view?usp=sharing)

**Comment by plakitkelly** [comment]
Steps:
1. Go to last page
2. Delete all language pairs

Actual: Table disappears [VIDEO](https://drive.google.com/file/d/1sI7i9L08-NBqW_ey5sdTqFfmPn3u6El_/view?usp=sharing)

**Comment by plakitkelly** [comment]
Update vendor table automatically after adding/removing the vendor.
Currently user has to refresh the page or use the searching field, click on kuva korraga 10/50,

**Comment by plakitkelly** [comment]
filtering by language directions doesn't work 


![Image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/200209ea-a8ef-4b79-bd54-0a85a9c3cb69)

Actual: empty table 

![Image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/c92b3daf-8e40-4f37-aa8a-f5d924538221)

**Comment by KaarelKa** [comment]
@MariusJulius Maybe you can go over info in this task as well. Some of the things look like they were dealt with under other tickets. Not sure if this ticket is still relevant or not (Didn't retest everything)

**Comment by MariusJulius** [comment]
2. remove pagination 100 option

**Comment by KaarelKa** [comment]
1. Fixed, only 1 sorting can be selected at a time
2. Removed
3. Works more or less as expected, except for error handling. Added better error handling here, possibly will do some extra improvements in January.

**Comment by plakitkelly** [comment]
@KaarelKa Filtering and sorting buttons are disabled? 
![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/d85863ef-ebef-4c46-974c-cee5e0313125)

**Comment by KaarelKa** [comment]
@plakitkelly this specific page is done according to design, without any filtering sorting:

Design:

<img width="876" alt="Screenshot 2023-12-28 at 13 07 23" src="https://github.com/keeleinstituut/tv-tolkevarav/assets/26221722/cd69fcdb-346c-4169-b696-b9cfdeed3d76">


The fixes done here apply more for the "Koondhinnakiri" which uses the same endpoint

**Comment by plakitkelly** [comment]
@KaarelKa Yeah, my bad. But in Koondhinnakiri view user can't remove language pair filtering.
1. Select any language pair filtering
2. Open again filtering modal
3. Remove checkmark and click save

Actual : Nothing happens, Filtering is still applied but user can select other language pair for filtering

**Comment by plakitkelly** [comment]
And I can't add new language pair after getting the price already exist error. Save button does nothing

**Comment by KaarelKa** [comment]
Note: The mentioned issues should be fixed, but error handling might still have some issues. Basically the main concern is that the form does not represent the same data structure that is used by BE and for the endpoint. This means the errors we show under different fields are often error with the same field, when looking at it from BE perspective and we cant definitively map all the errors to the correct fields. I'm hoping it works mostly correctly now though
